pkg/apis/seedmanagement/types_managedseed.go

```diff
@@ -0,0 +1,182 @@
+type ManagedSeedSpec struct {
+	// Shoot is the Shoot that will be registered as Seed.
+	Shoot *Shoot
+	// Seed describes the Seed that will be registered.
+	// Either Seed or Gardenlet must be specified.
+	Seed *SeedTemplateSpec
+	// Gardenlet specifies gardenlet deployment parameters and the GardenletConfiguration used to configure gardenlet.
+	Gardenlet *Gardenlet
+}
+
+// Shoot identifies the Shoot that will be registered as Seed.
+type Shoot struct {
+	// Name is the name of the Shoot that will be registered as Seed.
+	Name string
+	// APIServer specifies certain kube-apiserver parameters of the Shoot that will be registered as Seed.
+	APIServer *APIServer
+}
+
+// SeedTemplateSpec describes the data a Seed should have when created from a template.
+type SeedTemplateSpec struct {
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Specification of the desired behavior of the Seed.
+	Spec gardencorev1beta1.SeedSpec
+}
+
+// Gardenlet specifies gardenlet deployment parameters and the GardenletConfiguration used to configure gardenlet.
+type Gardenlet struct {
+	// Deployment specifies certain gardenlet deployment parameters, such as the number of replicas,
+	// the image, which bootstrap mechanism to use (bootstrap token / service account), etc.
+	Deployment *GardenletDeployment
+	// Config is the GardenletConfiguration used to configure gardenlet.
+	Config *gardenletconfigv1alpha1.GardenletConfiguration
+	// Bootstrap is the mechanism that should be used for bootstrapping gardenlet connection to the Garden cluster. One of ServiceAccount, Token.
+	// If specified, a service account or a bootstrap token will be created in the garden cluster and used to compute the bootstrap kubeconfig.
+	// If not specified, the gardenClientConnection.kubeconfig field will be used to connect to the Garden cluster.
+	GardenConnectionBootstrap *GardenConnectionBootstrap
+	// SeedConnection is the mechanism for gardenlet connection to the Seed cluster. Must equal ServiceAccount if specified.
+	// If not specified, the seedClientConnection.kubeconfig field will be used to connect to the Seed cluster.
+	SeedConnection *SeedConnection
+	// MergeParentConfig specifies whether the deployment parameters and GardenletConfiguration of the parent gardenlet
+	// should be merged with the specified deployment parameters and GardenletConfiguration. Defaults to false.
+	MergeParentConfig bool
+}
+
+// GardenletDeployment specifies certain gardenlet deployment parameters, such as the number of replicas,
+// the image, which bootstrap mechanism to use (bootstrap token / service account), etc.
+type GardenletDeployment struct {
+	// ReplicaCount is the number of gardenlet replicas. Defaults to 1.
+	ReplicaCount *int32
+	// RevisionHistoryLimit is the number of old gardenlet ReplicaSets to retain to allow rollback. Defaults to 10.
+	RevisionHistoryLimit *int32
+	// ServiceAccountName is the name of the ServiceAccount to use to run gardenlet pods.
+	ServiceAccountName string
```

Comment:
Optional string, should be a pointer, see comment in versioned API.
```suggestion
	ServiceAccountName *string
```

Reply:
OK